all-the-icons-dired.el

```diff
@@ -44,56 +46,80 @@
+  "Get all all-the-icons-dired overlays between BEG to END."
+  (cl-remove-if-not
+   (lambda (ov)
+     (overlay-get ov 'all-the-icons-dired-overlay))
+   (overlays-in beg end)))
+
+(defun all-the-icons-dired--overlays-at (pos)
+  "Get all-the-icons-dired overlays at POS."
+  (apply #'all-the-icons-dired--overlays-in `(,pos ,pos)))
+
+(defun all-the-icons-dired--remove-all-overlays ()
+  "Remove all `all-the-icons-dired' overlays."
+  (save-restriction
+    (widen)
+    (mapc #'delete-overlay
+          (all-the-icons-dired--overlays-in (point-min) (point-max)))))
+
+(defun all-the-icons-dired--refresh ()
   "Display the icons of files in a dired buffer."
-  (when (and (not all-the-icons-dired-displayed) dired-subdir-alist)
-    (setq-local all-the-icons-dired-displayed t)
-    (let ((inhibit-read-only t)
-	  (remote-p (and (fboundp 'tramp-tramp-file-p)
-                         (tramp-tramp-file-p default-directory))))
-      (save-excursion
-	(goto-char (point-min))
-	(while (not (eobp))
-	  (when (dired-move-to-filename nil)
-	    (let ((file (dired-get-filename 'verbatim t)))
-	      (unless (member file '("." ".."))
-		(let ((filename (dired-get-filename nil t)))
-		  (if (file-directory-p filename)
-		      (let* ((matcher (all-the-icons-match-to-alist file all-the-icons-dir-icon-alist))
-			     (icon (cond
-				    (remote-p
-				     (all-the-icons-octicon "file-directory" :v-adjust all-the-icons-dired-v-adjust :face 'all-the-icons-dired-dir-face))
-				    ((file-symlink-p filename)
-				     (all-the-icons-octicon "file-symlink-directory" :v-adjust all-the-icons-dired-v-adjust :face 'all-the-icons-dired-dir-face))
-				    ((all-the-icons-dir-is-submodule filename)
-				     (all-the-icons-octicon "file-submodule" :v-adjust all-the-icons-dired-v-adjust :face 'all-the-icons-dired-dir-face))
-				    ((file-exists-p (format "%s/.git" filename))
-				     (all-the-icons-octicon "repo" :v-adjust all-the-icons-dired-v-adjust :face 'all-the-icons-dired-dir-face))
-				    (t (apply (car matcher) (list (cadr matcher) :face 'all-the-icons-dired-dir-face :v-adjust all-the-icons-dired-v-adjust))))))
-			(insert (concat icon " ")))
-		    (insert (concat (all-the-icons-icon-for-file file :v-adjust all-the-icons-dired-v-adjust) " ")))))))
-	  (forward-line 1))))))
-
-(defun all-the-icons-dired--reset (&optional _arg _noconfirm)
+  (all-the-icons-dired--remove-all-overlays)
+  (save-excursion
+    (goto-char (point-min))
+    (while (not (eobp))
+      (let ((file (dired-get-filename 'verbatim t)))
+        (when file
+          (let ((icon (if (file-directory-p file)
+                          (all-the-icons-icon-for-dir file nil "")
+                        (all-the-icons-icon-for-file file :v-adjust all-the-icons-dired-v-adjust))))
+            (if (member file '("." ".."))
+                (all-the-icons-dired--add-overlay (point) "  ")
+              (all-the-icons-dired--add-overlay (point) (concat icon " "))))))
```

Comment:
What about using: `(setq-local tab-width 1)` 
And then `(concat icon "\t")`

It's a neat trick to make the icons aligned in the buffer. Taken from: https://github.com/seagle0128/.emacs.d/blob/master/lisp/init-dired.el#L82-L113

Reply:
Yes. I've tested. The trick is still works here. I will submit a PR.